src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/TriviaDataFactory.Analyzer.cs

```diff
@@ -73,11 +73,8 @@ public static AnalysisResult Between(SyntaxToken token1, SyntaxToken token2)
                         }
 
                         // Finally hit the first previous token with non-zero full width.
-                        if (previousToken.TrailingTrivia.Count > 0 &&
-                            previousToken.TrailingTrivia.Last().Kind() == SyntaxKind.EndOfLineTrivia)
-                        {
+                        if (previousToken.TrailingTrivia is [.., (kind: SyntaxKind.EndOfLineTrivia)])
```

Comment:
How does this work? What kind of pattern is `(kind: ...)`?

Reply:
Deconstruction? Do we have some extension Deconstruct method on trivia?

Reply:
yup.  we deconstruct SyntaxNode/Token/Trivia just so we can actually do reasonable 'kind' checks on them in our code.  It's powerful to as you can do things like:  `if (x is BinaryExpressionSyntax(SyntaxKind.AddExpression) binExpr)`  so you can get the strong typing of the variable, and the precise kind checks.  We've had it for around a year now i think?

Reply:
Also, the `(kind: ...` syntax is necessary, otherwise you have `is [.., (SyntaxKind.EndOfLineTrivia)]` and that becomes just a constant-parenthesized-pattern, not a deconstruct pattern.

Reply:
> (x is BinaryExpressionSyntax(SyntaxKind.AddExpression)

This is consistently the most confusing pattern for me to read, and one i never think of writing, though the IDE offers no help in writing it anyway 😝

Reply:
time to start using it more then :)